source/OpcodeInfoDatabase.cpp

```diff
@@ -52,166 +51,302 @@ bool OpcodeInfoDatabase::Load(const char* filepath)
+			if(!c["operator"].get_c_str().get(commandOperator))
 			{
-				continue; // opcode out of bounds
+				command.oper = commandOperator;
 			}
-			auto id = (uint16_t)idLong;
 
-			extension.opcodes.emplace(piecewise_construct, make_tuple(id), make_tuple(id, commandName));
-			auto& opcode = extension.opcodes.at(id);
+			// attributes
+			dom::element attr;
+			if (!c["attrs"].get(attr))
+			{
+				bool state;
+				if (!attr["is_condition"].get_bool().get(state)) { command.isCondition = state; }
+				if (!attr["is_constructor"].get_bool().get(state)) { command.isConstructor = state; }
+				if (!attr["is_nop"].get_bool().get(state)) { command.isNop = state; }
+			}
 
-			// read arguments info
-			dom::array inputArgs;
-			if (!c["input"].get_array().get(inputArgs))
+			// command params
+			auto LoadParams = [&](const char* obj, bool isInput)
 			{
-				for (auto& p : inputArgs)
+				dom::array args;
+				if (!c[obj].get_array().get(args))
 				{
-					const char* argName;
-					if (!p["name"].get_c_str().get(argName))
+					for (auto& p : args)
 					{
-						opcode.arguments.emplace_back(argName);
+						const char* argName;
+						if (!p["name"].get_c_str().get(argName))
+						{
+							auto& arg = command.arguments.emplace_back();
+							if (isInput) command.inputArguments++;
+
+							arg.name = argName;
+
+							const char* argType;
+							if (!p["type"].get_c_str().get(argType))
+							{
+								arg.typeName = argType;
+								arg.typeNameLower = argType;
+								CLEO::StringToLower(arg.typeNameLower);
+								arg.type = TypeFromName(arg.typeName);
+							}
+
+							const char* argSource;
+							if (!p["source"].get_c_str().get(argSource))
+							{
+								arg.source = argSource;
+								CLEO::StringToLower(arg.source);
+							}
+						}
 					}
 				}
-			}
-		}
-
-		if (!extension.opcodes.empty())
-		{
-			extensions[extension.name] = move(extension);
+			};
+			LoadParams("input", true);
+			LoadParams("output", false);
 		}
 	}
 
-	TRACE("Opcodes database version %s loaded from '%s'", version, filepath);
-	ok = true;
+	TRACE("Commands database version %s loaded from '%s'", version, filepath);
 	return true;
 }
 
-void OpcodeInfoDatabase::Clear()
+bool OpcodeInfoDatabase::LoadEnums(const char* filepath)
 {
-	ok = false;
-	extensions.clear();
-}
+	m_enums.clear();
 
-const char* OpcodeInfoDatabase::GetExtensionName(uint16_t opcode) const
-{
-	if (ok)
+	std::ifstream file(filepath, std::ios::in);
+	if (!file.is_open())
 	{
-		for (auto& entry : extensions)
+		TRACE("Failed to open enums database '%s' file", filepath);
+		return false;
+	}
+
+	bool inEnum = false;
+	Enum currEnum;
+	std::string line;
+	size_t lineCount = 0;
+	while (std::getline(file, line))
+	{
+		lineCount++;
+
+		CLEO::StringTrim(line);
+
+		if (line.empty()) continue;
+
+		if (!inEnum)
 		{
-			auto& extension = entry.second;
-			auto& opcodes = extension.opcodes;
+			if (CLEO::StringStartsWith(line, "enum ", false))
```

Comment:
use enums.json instead [0D5A](https://raw.githubusercontent.com/sannybuilder/library/refs/heads/master/sa/enums.json)

no need to parse enums.txt